lib/readyset.rb

```diff
@@ -93,8 +91,43 @@ def self.drop_cache!(name_or_id: nil, sql: nil)
   #
   # @param [Array<Object>] *sql_array the SQL array to be executed against ReadySet
   # @return [PG::Result]
-  def self.raw_query(*sql_array)
-    ActiveRecord::Base.establish_connection(Readyset::Configuration.configuration.database_url)
-    ActiveRecord::Base.connection.execute(ActiveRecord::Base.sanitize_sql_array(sql_array))
+  def self.raw_query(*sql_array) # :nodoc:
+    ActiveRecord::Base.connected_to(role: reading_role, shard: shard, prevent_writes: false) do
```

Comment:
What's the difference between setting `prevent_writes: false` and setting it to `true` on `.route`?

I'd imagine the default is to keep writes from happening on the replica.

Reply:
Or is it just for submitting the cache to Readyset?

Reply:
We can't prevent writes here because certain queries (e.g. `CREATE CACHE` or `DROP CACHE`) are identified as writes by the database adapter (since they start with `CREATE`/`DROP`) -- setting `prevent_writes: true` here would cause a runtime error when trying to run such queries. In the context of our private, undocumented `Readyset.raw_query` method, we always need to allow writes so we can send such queries to ReadySet without issue